test.py

```diff
@@ -563,16 +563,65 @@ def run_test(framework, cluster, repo_root):
         logger.info(msg, runtests_script, framework.name)
         raise CommandFailure(cmd_args)
 
+def setup_junit_xml():
+    """Try to download and import junit_xml"""
+    workdir = get_work_dir()
+    lib_dir = os.path.join(workdir, 'lib', 'python')
+    pip_cmd = ['pip', 'install', 'junit-xml', '-t', lib_dir]
```

Comment:
I'm still not clear. I said A or B and you said "yes".  Do you mean A and B?

It would be nice to have all the module downloading in one place.  However,  pip download is a common enough case of test failure, that I'd like to use junit to report it, which is why moved this to the very start of the run.

It could still make sense to do all the module fetching in one module with two different cases, but that would require further surgery to run_tests.py to readily achieve.  That said, I could readily introduce a utility module used both here and in configure_test_cluster.py and make run_tests.py use it later.  

If that covers your concerns then I'll do it.